src/engine/sidechain/engineshoutcast.cpp

```diff
@@ -98,7 +98,15 @@ EngineShoutcast::EngineShoutcast(ConfigObject<ConfigValue>* _config)
 EngineShoutcast::~EngineShoutcast() {
     m_pShoutcastEnabled->set(0);
     m_readSema.release();
-    wait(); // until the thread ends.
+
+    // Wait maximum ~4 seconds. User will get annoyed but
+    // if there is some network problems we let them settle
+    wait(4000);
+
+    // Signal user if thread doesn't die
+    if (isRunning()) {
```

Comment:
IMHO this should be an 
DEBUG_ASSERT_AND_HANDLE(!isRunning()) {

Since it is a workaround for pending bug, which is also bad if you try to disconnect during a Mixxx run.